src/CMakeLists.txt

```diff
@@ -40,11 +40,17 @@ if(CMAKE_CXX_COMPILER_ID MATCHES GNU)
     if(ESPP_LOCAL_ARCHITECTURE)
         target_compile_options(_espressopp PRIVATE "-march=native")
     endif()
+    if(ESPP_WERROR)
+        target_compile_options(_espressopp PRIVATE "-Werror")
+    endif()
```

Comment:
As it is the same block everywhere, could we just write it one time on the outside.

Reply:
I also thought about this. I do not know if we want to add another compiler in the future. The current way is clearly structured but verbose and a little bit of duplication. I can live with both.